### PR TITLE
More GSPLCD and GSPGPU functions.

### PR DESCRIPTION
**Pull request by joel16**
Added the following functions using the information provided on 3DBrew. I did test them to make sure they're working and they did.

Result GSPLCD_PowerOnAllBacklights(void); -> https://www.3dbrew.org/wiki/GSPLCD:PowerOnAllBacklights
Result GSPLCD_PowerOffAllBacklights(void); -> https://www.3dbrew.org/wiki/GSPLCD:PowerOffAllBacklights
Result GSPLCD_SetLedForceOff(bool state); -> https://www.3dbrew.org/wiki/GSPLCD:SetLedForceOff

Result GSPGPU_SetLedForceOff(bool state); -> https://www.3dbrew.org/wiki/GSPGPU:SetLedForceOff

### CODE REVIEW

File: libctru/include/3ds/services/gspgpu.h [hunk 1]
```diff
@@ -218,3 +218,8 @@ Result GSPGPU_UnregisterInterruptRelayQueue(void);
 /// Triggers a handling of commands written to shared memory.
 Result GSPGPU_TriggerCmdReqQueue(void);
 
+/**
+ * @brief Sets 3D_LEDSTATE to the input state value.
+ * @param state 0 = 3D LED enable, 1 = 3D LED disable.
+ */
+Result GSPGPU_SetLedForceOff(u8 state);
```
Comment (fincs) [inline libctru/include/3ds/services/gspgpu.h]: Should be `bool`

File: libctru/include/3ds/services/gsplcd.h [hunk 2]
```diff
@@ -32,6 +38,12 @@ Result GSPLCD_PowerOnBacklight(u32 screen);
 Result GSPLCD_PowerOffBacklight(u32 screen);
 
 /**
+ * @brief Sets 3D_LEDSTATE to the input state value.
+ * @param state 0 = 3D LED enable, 1 = 3D LED disable.
+ */
+Result GSPLCD_SetLedForceOff(u8 state);
```
Comment (fincs) [inline libctru/include/3ds/services/gsplcd.h]: Should be `bool`

File: libctru/source/services/gspgpu.c [hunk 3]
```diff
@@ -430,3 +430,15 @@ Result GSPGPU_RestoreVramSysArea(void)
 	return cmdbuf[1];
 }
 
+Result GSPGPU_SetLedForceOff(bool state)
+{
+	u32 *cmdbuf = getThreadCommandBuffer();
+
+	cmdbuf[0] = IPC_MakeHeader(0x1C,1,0); // 0x1C0040
+	cmdbuf[1] = state;
```
Comment (fincs) [inline libctru/source/services/gspgpu.c]: bool parameters are written as `state & 0xFF` to the cmdbuf

File: libctru/source/services/gsplcd.c [hunk 4]
```diff
@@ -52,6 +76,19 @@ Result GSPLCD_PowerOffBacklight(u32 screen)
 	return cmdbuf[1];
 }
 
+Result GSPLCD_SetLedForceOff(bool state)
+{
+	u32 *cmdbuf = getThreadCommandBuffer();
+
+	cmdbuf[0] = IPC_MakeHeader(0x13,1,0); // 0x130040
+	cmdbuf[1] = state;
```
Comment (fincs) [inline libctru/source/services/gsplcd.c]: Same here

**Comment by joel16** [comment]
Oh right, cause I was having a look at the am services and noticed you didn't do anything with the bool parameters so I left it as that. 
https://github.com/smealum/ctrulib/blob/master/libctru/source/services/am.c#L641

Also curious as to why casting to (Result) is required there?

File: libctru/source/services/gspgpu.c (hunk 3 above)
Comment (mtheall) [inline libctru/source/services/gspgpu.c]: No, that's only for output bools. These are both inputs.

File: libctru/include/3ds/services/gsplcd.h (hunk 2 above)
Comment (mtheall) [inline libctru/include/3ds/services/gsplcd.h]: I would also rename `state` to `off`

Comment (mtheall) [inline libctru/include/3ds/services/gsplcd.h]: Or `disable`

Comment (joel16) [inline libctru/include/3ds/services/gsplcd.h]: Ok, going to go with disable.

Comment (mtheall) [inline libctru/include/3ds/services/gsplcd.h]: cmdbuf is u32 and Result is s32.

**Pull request merged**